### PR TITLE
Adjust late penalty calculation

### LINKED ISSUE #214: backend: Late penalties count late days properly over weekends
@19mdavenport Shared the following relevant comment:
> When calculating late penalties, should Saturday and Sunday count as the equivalent of Friday or Monday? I always thought it would be Monday, but it looks like the autograder code treats it as Friday.

### Late TurnIn over Weekends
Here is a table of how we would expect the penalties to apply
| Date | Expected Days Late | Current Days Late |
|--------|--------|--------|
| Thursday (**Due**) | 0 | 0 |
| Friday | 1 | 1 |
| Saturday | 2 | 1 |
| Sunday | 2 | 1 |
| Monday | 2 | 2 | 
| Tuesday | 3 | 3 |

### Late TurnIn over Holidays
Here is a table of how we would expect the penalties to apply
| Date | Expected Days Late | Current Days Late |
|--------|--------|--------|
| Thursday (**Due**) | 0 | 0 |
| Friday | 1 | 1 |
| Saturday | 2 | 1 |
| Sunday | 2 | 1 |
| Monday (**Holiday**) | 2 | 2 | 
| Tuesday | 2 | 3 |

### PR DESCRIPTION
**Pull request by 19mdavenport**
This adjusts the late penalty calculations so that weekends are counted as an extra day late (Saturday and Sunday matching the next Monday's calculation rather than the previous Friday's) resolves #214

### CODE REVIEW

**Comment by frozenfrank** [comment]
@pawlh This one is ready for review, and I have another one coming soon with additional improvements

**Review by frozenfrank — APPROVED** [review]
The adjustments that @19mdavenport made work really well and pass all of the tests.

I provided a set of tests that validate the inputs with diverse timezones, and ensure that the days are counted as expected over the weekends.

File: src/main/java/edu/byu/cs/util/DateTimeUtils.java [hunk 1]
```diff
@@ -34,10 +34,10 @@ public static int getNumDaysLate(ZonedDateTime handInDate, ZonedDateTime dueDate
         int daysLate = 0;
 
         while (handInDate.isAfter(dueDate)) {
-            if (handInDate.getDayOfWeek().getValue() < 6 && !isPublicHoliday(handInDate)) {
+            if (dueDate.getDayOfWeek().getValue() < 6 && !isPublicHoliday(dueDate)) {
                 daysLate++;
             }
-            handInDate = handInDate.minusDays(1);
+            dueDate = dueDate.plusDays(1);
```
Comment (pawlh) [inline src/main/java/edu/byu/cs/util/DateTimeUtils.java:40]: Ahhh, I see why my code wasn't working. Nice fix!

File: src/test/java/edu/byu/cs/util/DateTimeUtilsTest.java
Comment (pawlh) [inline src/test/java/edu/byu/cs/util/DateTimeUtilsTest.java:1]: Tests! 💯

**Review by pawlh — APPROVED** [review]

**Pull request merged**